### ISSUE TITLE
Question about node_indicator

### ISSUE THREAD

**Issue by Ghostish** [XinyiZ001/CapsGNN#1]
Hi, 
Thanks for providing the codes.
I am a bit confused about the `node_indicator` in the codes. In the codes it seems to be a mask for choosing nodes but I couldn't find any clues about it in the paper. 
And it's computed by:
`node_indicator = tf.reduce_max(batch_adj_mats, axis=-1, keep_dims=True)  # (?, N, 1)` 
Is it used to indicate nodes that have connections to others?
And i didn't find any codes that adds self-connections to the graph: `A + I` . Are all the nodes in the graphs self-connected? If so, the `node_indicator`  will be always full of "1"s, making no sense.

I am looking forward for you reply.

**Comment by XinyiZ001** [comment]
Hi,

Sorry for the late reply.
Actually, `node_indicator` is just used for coding, so it is not mentioned in the paper.

When we generate each batch of data, since the size of each graph is different, we need to pad the adjacency matrix of some graphs with `0`.  `node_indicator` is used as a node mask for each batch of data.

For example, 
batch size: 2
batch data (adjacency matrix):
```
|1110|   |1110|
|1110|   |1101|
|1110|   |1010|
|0000|   |0101|
```
correspond `node_indicator`:
```
|1|   |1|
|1|   |1|
|1|   |1|
|0|   |1|
```
we just use `node indicator` to indicate the existence of nodes of each batch.

pls feel free to contact us if you still have questions.

**Comment by Ghostish** [comment]
Thanks for the reply,  I totally got that idea 👍
